### ISSUE TITLE
Task: Clean column names and naming conventions

### ISSUE THREAD

**Issue by sommergeo** [sommergeo/roadDB#3]
## TLDR
Rename column name `Locality ID` to `locality_id`.

## Problem
It is very helpful that internal naming conventions , like `cm_assemblages_locality_idlocality` are renamed to user-friendly short names like `Locality ID`. 
However, the space character within the column name is just as problematic as other special characters (e.g. `+`, `-`, etc.).
If a user wants to access a specific column, the command `result$Locality ID` results$`Locality ID` will not work due to issues with the space character. Instead, the cumbersome `results$'Locality ID'` must be used. 

## Solution
To overcome this, it is proposed to use a naming convention throughout the package **avoiding spaces and special characters** and using the [**snake_case**](https://en.wikipedia.org/wiki/Snake_case) (as already applied to the function names, e.g. `road_get_localities()`).
